standalone-examples/src/main/java/io/pravega/example/streamcuts/StreamCutsExample.java

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package io.pravega.example.streamcuts;
+
+import com.google.common.collect.Lists;
+import io.pravega.client.ClientFactory;
+import io.pravega.client.admin.ReaderGroupManager;
+import io.pravega.client.admin.StreamManager;
+import io.pravega.client.batch.BatchClient;
+import io.pravega.client.batch.SegmentRange;
+import io.pravega.client.batch.StreamSegmentsIterator;
+import io.pravega.client.stream.Checkpoint;
+import io.pravega.client.stream.EventRead;
+import io.pravega.client.stream.EventStreamReader;
+import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.ReaderConfig;
+import io.pravega.client.stream.ReaderGroup;
+import io.pravega.client.stream.ReaderGroupConfig;
+import io.pravega.client.stream.ReinitializationRequiredException;
+import io.pravega.client.stream.ScalingPolicy;
+import io.pravega.client.stream.Stream;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.client.stream.StreamCut;
+import io.pravega.client.stream.impl.JavaSerializer;
+import java.io.Closeable;
+import java.net.URI;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.function.Consumer;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class StreamCutsExample implements Closeable {
+
+    public static final int maxEventsPerDay = 3;
+    private static final String eventSeparator = ":";
+    private static final String streamSeparator = "-";
+
+    private int numStreams;
+    private int numEvents;
+    private URI controllerURI;
+    private String scope;
+    private ScheduledExecutorService executor;
+    private StreamManager streamManager;
+    private List<String> myStreamNames = new ArrayList<>();
+    private Map<String, SimpleEntry<Integer, Integer>> perDayEventIndex = new LinkedHashMap<>();
+
+    public StreamCutsExample(int numStreams, int numEvents, String scope, URI controllerURI) {
+        this.numStreams = numStreams;
+        this.numEvents = numEvents;
+        this.controllerURI = controllerURI;
+        this.scope = scope;
+        streamManager = StreamManager.create(controllerURI);
+        executor = new ScheduledThreadPoolExecutor(1);
+    }
+
+    /**
+     * A {@link StreamCut} is an offset that indicates an event boundary. With a {@link StreamCut}, users can instruct
```

Comment:
This definition is not crispy, a stream cut is a collection of offsets, one for each open segment of the stream. Perhaps you are trying to discuss it in the context of this particular example, in which case it is best to start with something like, `In this example,....` to scope it out..

Reply:
You are right, I improved the `streamcut` definition following your guidelines.